internal/clients/http/azure/create_vm.go

```diff
@@ -0,0 +1,395 @@
+
+	subnet, err := c.createSubnets(ctx, resourceGroupName, vnetName, subnetName)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create subnet")
+		logger.Error().Err(err).Msgf("cannot create subnet")
+		return nil, err
+	}
+	logger.Trace().Msgf("Using subnet id=%s", *subnet.ID)
+
+	publicIP, err := c.createPublicIP(ctx, location, resourceGroupName, publicIPName)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create public IP address")
+		logger.Error().Err(err).Msgf("cannot create public IP address")
+		return nil, err
+	}
+	logger.Trace().Msgf("Using public IP address id=%s", *publicIP.ID)
+
+	// network security group
+	nsg, err := c.createNetworkSecurityGroup(ctx, location, resourceGroupName, nsgName)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create network security group")
+		logger.Error().Err(err).Msgf("cannot create network security group")
+		return nil, err
+	}
+	logger.Trace().Msgf("Using network security group id=%s", *nsg.ID)
+
+	networkInterface, err := c.createNetworkInterface(ctx, location, resourceGroupName, subnet, publicIP, nsg, nicName)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create network interface")
+		logger.Error().Err(err).Msgf("cannot create network interface")
+		return nil, err
+	}
+	logger.Trace().Msgf("Using network interface id=%s", *networkInterface.ID)
+
+	virtualMachine, err := c.createVirtualMachine(ctx, location, resourceGroupName, networkInterface, imageID, sshKeyBody, vmName, diskName)
+	if err != nil {
+		span.SetStatus(codes.Error, "cannot create virtual machine")
+		logger.Error().Err(err).Msgf("cannot create virtual machine")
+		return nil, err
+	}
+	logger.Debug().Msgf("Created virtual machine id=%s", *virtualMachine.ID)
+
+	return virtualMachine, nil
+}
+
+func (c *client) createResourceGroup(ctx context.Context, name string, location string) (*armresources.ResourceGroup, error) {
+	resourceGroupClient, err := c.newResourceGroupsClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armresources.ResourceGroup{
+		Location: ptr.To(location),
+		Tags:     map[string]*string{"sample-rs-tag": ptr.To("sample-tag")}, // resource group update tags
+	}
+
+	resp, err := resourceGroupClient.CreateOrUpdate(ctx, name, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resp.ResourceGroup, nil
+}
+
+func (c *client) createVirtualNetwork(ctx context.Context, resourceGroupName string, name string) (*armnetwork.VirtualNetwork, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "createVirtualNetwork")
+	defer span.End()
+
+	vnetClient, err := c.newVirtualNetworksClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	parameters := armnetwork.VirtualNetwork{
+		Location: to.Ptr(location),
+		Properties: &armnetwork.VirtualNetworkPropertiesFormat{
+			AddressSpace: &armnetwork.AddressSpace{
+				AddressPrefixes: []*string{
+					to.Ptr("10.42.0.0/16"),
+				},
```

Comment:
I think for V1 this could be fine, but I suggest to select a "random constant" subnet that is the most unlikely to hit existing subnets. I have a gut feeling that 10.0.0.0 space will be the most popular one, it is also the one that Azure documentation recommends. But there are others:

```
The Internet Assigned Numbers Authority (IANA) has reserved the
   following three blocks of the IP address space for private internets:

     10.0.0.0        -   10.255.255.255  (10/8 prefix)
     172.16.0.0      -   172.31.255.255  (172.16/12 prefix)
     192.168.0.0     -   192.168.255.255 (192.168/16 prefix)
```

How about selecting `172.16.0.0/16` prefix that gives:

![image](https://user-images.githubusercontent.com/49752/214555066-465ba219-bd17-4a08-b0e0-9a7aed416e14.png)

That is 64k hosts (16 bits) for each which is good enough for most customers, yet, we can select one of the 172.16 - 172.31 subnets randomly (constant). For example `172.22.0.0/16` is a nice one to remember.

Now this is a wild guess, but I believe this one is less likely to collide with customers existing networks. Of course, this is a temporary solution until we implement templates.

Reply:
For the record, AWS VPC has a default range `172.31.0.0/16`: https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html